packages/lab/src/pill-next/PillNext.tsx

```diff
@@ -0,0 +1,80 @@
+import {
+  forwardRef,
+  MouseEvent,
+  PropsWithChildren,
+  KeyboardEvent,
+  ComponentPropsWithoutRef,
+} from "react";
+import clsx from "clsx";
+import { useWindow } from "@salt-ds/window";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { makePrefixer, useButton } from "@salt-ds/core";
+import pillCss from "./PillNext.css";
+
+export type PillClickEvent =
+  | MouseEvent<Element, globalThis.MouseEvent>
+  | KeyboardEvent<HTMLDivElement>;
+
+export interface PillNextProps extends ComponentPropsWithoutRef<"div"> {
+  /* If true the pill will be disabled */
+  disabled?: boolean;
+  /* Pass an element to render an icon descriptor on the left of the label */
+  icon?: React.ReactNode;
+  /* Pass a callback function to make the pill clickable */
+  onClick?: (event: PillClickEvent) => void;
+}
+
+const withBaseName = makePrefixer("saltPillNext");
+
+export const PillNext = forwardRef<
+  HTMLDivElement,
+  PropsWithChildren<PillNextProps>
+>(function PillNext(
+  { onClick, children, className, icon, disabled, ...restProps },
+  ref
+) {
+  const targetWindow = useWindow();
+  useComponentCssInjection({
+    testId: "salt-pill",
+    css: pillCss,
+    window: targetWindow,
+  });
+  const {
+    buttonProps: { disabled: disabledAttribute, ...buttonProps },
+    active,
+  } = useButton<HTMLDivElement>({
+    disabled,
+    onClick,
+    onKeyUp: (event) => {
+      if (disabled) return;
+      if (event.key === "Enter" || event.key === " ") {
+        onClick?.(event);
```

Comment:
This changes what onClick means if it is called during onKeyUp. Shouldn't this just happen in onClick?

Reply:
This is emulating button behaviour since useButton doesn't

Reply:
It's not really emulating Button behaviour though? onClick is being called on the "keyup" event

Reply:
Ok, true, wrong event. 😅  Should be keydown (with possibly repeating on long press). How about this... I will remove all of this and use the native button element and we can revisit this conversation if we want nested close button and possibly to implement https://github.com/jpmorganchase/salt-ds/issues/1964